vortex-array/src/encoding.rs

```diff
@@ -95,3 +95,99 @@ pub trait ArrayEncodingExt {
 pub trait ArrayEncodingRef {
     fn encoding(&self) -> EncodingRef;
 }
+
+#[doc = "Encoding ID constants for all Vortex-provided encodings"]
+#[allow(dead_code)]
+pub mod ids {
```

Comment:
really wanted to avoid this being in the main package but not sure if there's a way that this would be useful otherwise

Reply:
we can put it upstream of `vortex-array`, but it's not particularly useful if we can't reference it in the ~half of encodings that are in that crate

Reply:
IMO just put it in the README. Document the reserved range for Vortex, and the range for "custom" extensions.